### ISSUE TITLE
FR: add the ability to pass abort signal to httpsCallable

### ISSUE THREAD

**Issue by makoffd** [firebase/firebase-js-sdk#3055]
### Describe your environment

  * Operating System version: macOs Catalina 10.15.4
  * Browser version: Google Chrome 81.0.4044.129
  * Firebase SDK version: 7.14.3
  * Firebase Product: functions

### Describe the problem
It's impossible to cancel the `httpsCallable` call in response to some user event e.g. button click. 
The proposed solution is to pass down the abort signal to the `fetch` invocation as described in [MDN AbortSignal docs](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal). The `signal` parameter could be a part of `HttpsCallableOptions`.

#### Relevant Code:
```javascript
interface HttpsCallableOptions {
    timeout?: number;
    signal?: AbortSignal;
}

const controller = new AbortController();
const abortSignal = controller.signal;

functions.httpsCallable('myFirebaseFunction', {
    signal: abortSignal
});
```

**Comment by somq** [comment]
Any update on this one?

**Comment by its0xFUL** [comment]
This is more relevant with Angular than ever, with the new `resource` functionality. Please consider implementing this.

**Comment by tmcdo1** [comment]
Yes, this would be super useful for some use-cases to just cancel blocking network calls, even if the underlying firebase functions can't actually be cancelled once running.
